README.md

```diff
@@ -118,7 +118,113 @@ A signal can contain several actions. For example, if your app subscribes to the
 
 You can define a payload used by all actions, or for a specific action. When you define a method called "payload" this method will be used by all actions that don't have an action specific payload defined. If you wish, you can define an action specific payload by defining a method with the action name followed by `_payload`. As an example, see the create specific payload defined in the `create_payload` method in the example.
 
-Inside the signal class, you will have access to the instance variable `@object_id` which represents the ID of the "object" of the signal (Lesson in this case). You also have access to `object` which will give you the initialized object for that ID. You can customize your signal further as you wish, for example in this signal we've created a private method to alias `object` as `lesson` and using that in our payload definitions.
+Inside the signal class, you will have access to the method `object_id` which represents the ID of the "object" of the signal (Lesson in this case). You also have access to `object` which will give you the initialized object for that ID. You can customize your signal further as you wish, for example in this signal we've created a private method to alias `object` as `lesson` and using that in our payload definitions. You also have access to the `context` method to access any context passed when the signal is sent.
```

Comment:
Was the change from `@object_id` to `object_id` in the docs above intentional? Just wanted to double check :-)

Reply:
Ah yeah... before the pattern that I was using was to referenced a instance variable from the parent. Our Reek check in source level pointed out that it's not a great idea to do do that, because it's both unclear what the variable represents in the child class and makes the parent class resistant to change (because now, if you change an instance variable in the parent class it can have unintended consequences that are surprising because you wouldn't generally need to look outside of a class before changing an instance variable.)

The better solution is to make an actually public method that is a part of the class's API that you can call and access information though. That approach makes it easier to change the parent class, easier to document, and clearer that calling that method is a part of the class's public API https://github.com/troessner/reek/blob/master/docs/Instance-Variable-Assumption.md

So, based on all that it's a change I'm making to how this pattern is documented and how I use it.